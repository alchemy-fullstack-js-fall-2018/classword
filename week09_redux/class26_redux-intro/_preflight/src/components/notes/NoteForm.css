@import '../../styles/variables.css';

.noteForm {
  label {
    display: inline-block;
    width: 5em;
    margin-right: 1em;
  }

  input, textarea {
    width: 100%;
  }

  button {
    display: block;
    width: 100%;
    background-color: var(--primary);
    padding: 0.5em;
    color: var(--lightWhite);
    border: 1px solid #000;
    cursor: pointer;
  }
}
