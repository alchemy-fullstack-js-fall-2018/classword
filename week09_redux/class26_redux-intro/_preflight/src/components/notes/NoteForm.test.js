import React from 'react';
import { shallow } from 'enzyme';
import NoteForm from './NoteForm';

jest.mock('../../routes/routes.js');

describe('NoteForm', () => {
  let handleSubmit = jest.fn();
  let wrapper;
  beforeEach(() => {

    wrapper = shallow(<NoteForm onSubmit={handleSubmit} />);
  });

  it('renders a NoteForm component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('displays a note title', () => {
    wrapper.find('input').simulate('change', { target: { name: 'title', value: 'hi there' } });
    wrapper.find('input').simulate('change', { target: { name: 'body', value: 'This is a note' } });

    wrapper.find('form').simulate('submit', { preventDefault: jest.fn() });
    expect(handleSubmit.mock.calls[0][0]).toEqual({
      title: 'hi there',
      body: 'This is a note'
    });
  });
});
