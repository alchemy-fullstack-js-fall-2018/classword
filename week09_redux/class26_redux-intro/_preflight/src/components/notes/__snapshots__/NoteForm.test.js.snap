// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteForm renders a NoteForm component 1`] = `
<form
  className="noteForm"
  onSubmit={[Function]}
>
  <fieldset>
    <legend>
      Create a note
    </legend>
    <p>
      <label>
        Title
      </label>
      <input
        name="title"
        onChange={[Function]}
        value=""
      />
    </p>
    <p>
      <label>
        Body
      </label>
      <textarea
        name="body"
        onChange={[Function]}
        value=""
      />
    </p>
    <button
      type="submit"
    >
      Add Note!
    </button>
  </fieldset>
</form>
`;
